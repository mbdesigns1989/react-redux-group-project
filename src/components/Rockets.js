/* eslint-disable no-unused-vars */
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockesSlice';

const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets.entities);

  useEffect(() => {
    dispatch(fetchRockets());
  }, []);

  return (
    <div>
      <p>Hi</p>
    </div>
  );
};

export default Rockets;
