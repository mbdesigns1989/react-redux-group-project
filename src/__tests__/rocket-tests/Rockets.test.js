import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rockets from '../../components/Rockets';
import store from '../../redux/configureStore';

describe('Check if all components are rendered', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Rockets /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
