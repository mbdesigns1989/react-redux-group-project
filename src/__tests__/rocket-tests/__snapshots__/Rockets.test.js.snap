// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if all components are rendered renders correctly 1`] = `<div />`;
